services/utils.js

```diff
@@ -0,0 +1,75 @@
+const bcrypt = require("bcrypt");
+const EMAIL_REGEX =
+  /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+const isValidEmailSyntax = (email_address) => EMAIL_REGEX.test(email_address);
+
+const isValidPassword = (password) =>
+  password.length > 8 &&
+  password.length < 20 &&
+  /[A-Z]/.test(password) &&
+  /[a-z]/.test(password) &&
+  /\d/.test(password) &&
+  /[!@#$%^&*()_+\-=[\]{};':"\\|,.<>/?]/.test(password);
+
+const composite = (base, salt, pepper) => `${salt}${base}${pepper} `;
+
+const compare = async (storedPassword, userSalt, input, RANGE) => {
+  for (let i = 0; i < RANGE; i++) {
+    let cmp_res = await bcrypt.compare(
+      composite(input, userSalt, i),
```

Comment:
Use camel case please, also your variable name is not so clear (Is cmp referred to compare? to composite?